src/test/java/teammates/test/cases/webapi/UpdateFeedbackResponseActionTest.java

```diff
@@ -158,7 +215,273 @@ public void testExecute_instructorFeedbackSubmissionMcqGenerateOptionsForTeams_s
+        updateRequest.setQuestionType(FeedbackQuestionType.MCQ);
+        updateRequest.setRecipientIdentifier(student5InCourse1.getEmail());
+        FeedbackMcqResponseDetails newDetails = new FeedbackMcqResponseDetails();
+        newDetails.setAnswer("TEAM_NOT_EXIST");
+        updateRequest.setResponseDetails(newDetails);
+
+        String[] studentAccessOtherTeamParams = new String[] {
+                Const.ParamsNames.INTENT, Intent.STUDENT_SUBMISSION.toString(),
+                Const.ParamsNames.FEEDBACK_RESPONSE_ID, typicalResponse2.getId(),
+        };
+
+        UpdateFeedbackResponseAction a = getAction(updateRequest, studentAccessOtherTeamParams);
+        assertThrows(UnauthorizedAccessException.class, a::checkAccessControl);
+    }
+
+    @Test
+    protected void testAccessControl_instructorAccessOtherGiver_inaccessible() {
+
+        ______TS("Instructor intends to access other person's response, should not be accessible");
+
+        loginAsInstructor(instructor2OfCourse1.getGoogleId());
+
+        assertEquals(FeedbackParticipantType.SELF,
+                logic.getFeedbackQuestion(typicalResponse3.getFeedbackQuestionId()).getGiverType());
+        assertNotEquals(instructor2OfCourse1.getEmail(), typicalResponse3.getGiver());
+
+        // send update request
+        FeedbackResponseUpdateRequest updateRequest = new FeedbackResponseUpdateRequest();
+        updateRequest.setQuestionType(FeedbackQuestionType.MCQ);
+        updateRequest.setRecipientIdentifier(instructor2OfCourse1.getEmail());
+        FeedbackMcqResponseDetails newDetails = new FeedbackMcqResponseDetails();
+        newDetails.setAnswer("TEAM_NOT_EXIST");
+        updateRequest.setResponseDetails(newDetails);
+
+        String[] instructorAccessOtherPersonParams = new String[] {
+                Const.ParamsNames.INTENT, Intent.INSTRUCTOR_SUBMISSION.toString(),
+                Const.ParamsNames.FEEDBACK_RESPONSE_ID, typicalResponse3.getId(),
+        };
+
+        UpdateFeedbackResponseAction a = getAction(updateRequest, instructorAccessOtherPersonParams);
+        assertThrows(UnauthorizedAccessException.class, a::checkAccessControl);
     }
 
+    @Test
+    protected void testAccessControl_instructorAccessOwnResponse_accessible() {
+
+        ______TS("Instructor intends to access own response, should be accessible");
+
+        loginAsInstructor(instructor1OfCourse1.getGoogleId());
+
+        assertEquals(instructor1OfCourse1.getEmail(), typicalResponse3.getGiver());
+
+        // send update request
+        FeedbackResponseUpdateRequest updateRequest = new FeedbackResponseUpdateRequest();
+        updateRequest.setQuestionType(FeedbackQuestionType.MCQ);
+        updateRequest.setRecipientIdentifier(typicalResponse3.getRecipient());
+        FeedbackMcqResponseDetails newDetails = new FeedbackMcqResponseDetails();
+        newDetails.setAnswer("TEAM_NOT_EXIST");
+        updateRequest.setResponseDetails(newDetails);
```

Comment:
I think it would be good to enwrap this code into a helper method, since you're using it a lot. You can add things like recipient identifier as parameters and reuse the method instead of copy and pasting this.